### DISCUSSION TITLE
I can't get memray to output with `--aggregate`

### DISCUSSION THREAD

**Discussion by ooliver1** [bloomberg/memray#440] [Q&A]
I'm running memray inside docker with `python -m memray run --aggregate -o memray/memray.$$.bin -m vibr`, memray being a docker volume.

It creates a blank file just fine, but once I terminate with `SIGINT`, nothing happens, then with `SIGTERM` it never outputs properly. I managed to get it to output properly once but didn't have the volume to catch it at all. I can only do this every 3 days because that's when the seemingly memory leak occurs (in production only :sweat_smile:).

**Comment by pablogsal** [comment]
Hi @ooliver1 and thanks for opening a discussion. Could you please describe a bit better what you are doing exactly and maybe adding a reproducer here so we can investigate? Otherwise there is no way we can assist.

Also, note this warning here:

https://bloomberg.github.io/memray/run.html#aggregated-capture-files

> Also, note that if the process is killed before tracking ends (for instance, by the Linux OOM killer), then the process will die before it finishes calculating its statistics, and so no useful information is ever written to the capture file. With the default file format, the capture file is usually still usable even if the process crashes or is killed, but with the aggregated file format it is not.

This means that if your process terminates abruptly (which may happen with `SIGINT` or `SIGTERM` under some circumstances like with custom handlers, you may not getting any output at all.

**Reply by ooliver1** [reply]
Seems to just be that I can't get the process to exit cleanly, something to debug on my side, sorry.
